### PR TITLE
Add Short Code field (fixes #8099)

### LINKED ISSUE #8099: Add Customer Short Code to Customer
- [ ] Add 3-digit short code that can be used in many ways for the customer
- [ ] Create auto-rendering function based on customer name such as Frappe Inc. -> FRI
- [ ] Blacklist certain offensive names

### PR DESCRIPTION
**Pull request by kickapoo**
Please @nabinhait  review :)

### CODE REVIEW

File: erpnext/selling/doctype/customer/customer.py [hunk 1]
```diff
@@ -231,3 +250,48 @@ def get_credit_limit(customer, company):
 		credit_limit = frappe.db.get_value("Company", company, "credit_limit")
 
 	return flt(credit_limit)
+
+
+@frappe.whitelist()
+def short_code_generator(customer_name):
+	new_sc = ''
```
Comment (nabinhait) [inline erpnext/selling/doctype/customer/customer.py]: put this in short_code.py (files this long should be separated)

**Comment by nabinhait** [comment]
- [ ] How is short code used? This is just creating the short code, not linked to any workflow yet
- [ ] Test cases?

A broader discussion, can we do without short codes completely? Is this global search in v8 is pretty good!

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Comment by kickapoo** [comment]
Short Code is self explained. Customer has a short code like: Frappe Inc --> FRI. 
Please @nabinhait provide documentation file path so @dottenbr can take over and add the proper explanation. 
I will add Test Cases. 
@nabinhait Blacklist can't be bootstraped, mayde someone want KKK as short code. Its up to the "someone".

**Comment by rmehta** [comment]
@kickapoo  we still need workflow linked (where is this used???) No point having just made a feature that is unconnected.

**Comment by kickapoo** [comment]
@dottenbr  please take over on @rmehta comment.

**Comment by dottenbr** [comment]
Customer Short Code is used for searching in quotations, sales orders. It will also be used at ESO to generate assembly numbers (item numbers). If you don't see this as useful to others we can make it a customization. We should decide these things before we implement it though.

**Comment by rmehta** [comment]
This point is not whether its useful, whether its complete.

**Comment by kickapoo** [comment]
@rmehta if you aren't donna use it there is not reason to complete it and spend time on it. I can write the Unit Tested that are requested for this.

So please clearify the final call with won't merge flag or not so we can schedule our eso-internal features.

**Comment by rmehta** [comment]
Its not about me using it. At this point, you are not using it either 😅

**Comment by dottenbr** [comment]
It is not complete but we have a path to how to use it. We are using it 100% and we need it but we don't have it because we designed it as a contribution rather than a customization.

Do you want this in or out @rmehta?

Why was this closed @kickapoo?

**Comment by nabinhait** [comment]
Closing this as per https://github.com/frappe/erpnext/wiki/Pull-Request-Guidelines. Please reopen after implementing workflow.

**Pull request closed without merging**